### PR TITLE
Zombie Mode

### PR DESCRIPTION
**Pull request by Super3222**
### Intent of your Pull Request

Ports over Kn0ss0s's zombie mode, and some of the changes that came with it. For better or worse. I'm still leaving parts behind, so I present you my bullet point of amazing facts to take in while waiting for this mode to be finished.

- I want to finish Vampire Mode & Xenomorph Mode before giving this my full intention. I've made `zombie.dm` compilable, so that's a start.

I'll do classes later.

#### Changelog

:cl:
rscadd: Zombie Mode is now (hopefully) in rotation.
rscadd: Zombie Mode starts with a certain amount of crewmembers being infected with the deadly zombie virus.
rscadd: As a zombie, your only weak spot is your head. Damage anywhere else to your body is negated for stun damage which will freeze you in place.
rscadd: As a human, understand that the cure to the rage virus is randomized! You should try to bunker down with your fellow co-workers, and find a way to combat the deadly infection.
rscadd: The round ends when there's either 0 humans or the shuttle lands at centcom. If the shuttle docks at centcom with 0 "living" zombies on board, then the crew will win. Otherwise, the zombies win.
/:cl:

### CODE REVIEW

**Comment by Mrcreepssss** [comment]
WALKING DEEEEEAAAADDDDD isnt really dead

File: code/datums/diseases/transformation.dm [hunk 1]
```diff
@@ -239,4 +239,53 @@
...
-	new_form = /mob/living/simple_animal/hostile/morph
\ No newline at end of file
+	new_form = /mob/living/simple_animal/hostile/morph
+
+
+/datum/disease/transformation/rage_virus
+	name = "Rage Virus"
+	//cure_text = "Whiskey"
+	//cures = list("whiskey")
+	spread_text = "zombie Bites"
```
Comment (ShadowDeath6) [inline code/datums/diseases/transformation.dm:249]: caps?

File: code/modules/mob/living/carbon/zombie/zombie.dm [hunk 2]
```diff
@@ -391,9 +375,8 @@
 /mob/living/carbon/human/zombie/put_in_hand_check(obj/item/W)
 	if(istype(W, /obj/item/weapon/card))
 		return 1
-
 	if(istype(W, /obj/item/weapon) || istype(W, /obj/item/clothing/suit/armor) || istype(W, /obj/item/clothing/suit/bio_suit) || istype(W, /obj/item/clothing/head/helmet))
-		src << "You Zombified brain struggles to comprehend picking up the [W]. Somehow it is just too complicated to operate."
+		src << "You brain struggles to comprehend picking up the [W]. Somehow it is just too complicated to operate."
```
Comment (ShadowDeath6) [inline code/modules/mob/living/carbon/zombie/zombie.dm:217]: me brain?

File: code/datums/diseases/transformation.dm (hunk 1 above)
Comment (Super3222) [reply code/datums/diseases/transformation.dm:249]: it isn't me this time. i swear.

File: code/modules/mob/living/carbon/zombie/zombie.dm (hunk 2 above)
Comment (matskuman5) [reply code/modules/mob/living/carbon/zombie/zombie.dm:217]: me am smart?

**Comment by ShadowDeath6** [comment]
I think the organ is still a cleaner way to do it

**Comment by Super3222** [comment]
The organ will make it harder to do classes.

**Comment by ShadowDeath6** [comment]
organ/zombie/subtype

**Comment by Super3222** [comment]
What I intend is a little more advanced.

**Comment by Super3222** [comment]
- EXPERIMENTAL: Zombies only take damage when the weapon/projectile is aimed at their head, otherwise they will get stun (frozen). Dont forget that zombies divide their stun damage by 3.
It takes 2 zombies to break open an airlock. The damage they do take to the head is multiplied by a factor of 1.5
- Zombies now slam into windows.
- No longer uses Kn0sses zombie sprites.
- Zombie attack paw now speeds up the virus stages.
- Zombie mobs get the infectious zombie organ. Has nothing to do with the rage virus - it heals them.
- Added a new variable to check_shields(), one for which limb is being targeted. 
- Removes a bunch of dead code in zombie.dm, and stuff that I don't want there.

**Comment by Super3222** [comment]
- EXPERIMENTAL: Zombies only take damage when the weapon/projectile is aimed at their head, otherwise they will be stunned (frozen). Dont forget that zombies divide their stun damage by 3. The damage they do take to the head is multiplied by a factor of 1.5
- It takes 2 zombies to break open an airlock. 
- Zombies now slam into windows.
- No longer uses Kn0sses zombie sprites.
- Zombie attack paw now speeds up the virus stages.
- Zombie mobs get the infectious zombie organ. Has nothing to do with the rage virus - it heals them.
- Added a new variable to check_shields(), one for which limb is being targeted. 
- Removes a bunch of dead code in zombie.dm, and stuff that I don't want there.

**Comment by ShadowDeath6** [comment]
new commit is the :+1:

**Comment by Super3222** [comment]
TODO:
- Virus randomization

**Comment by Lightning38088** [comment]
Can you make it possible for them to get around gliders or some how break them? i've seen people use gliders to surrond their base, making it Impossible for the zombies to get in.

**Pull request closed without merging**